Detectors/CTP/simulation/src/Digitizer.cxx

```diff
@@ -116,11 +122,28 @@ std::vector<CTPDigit> Digitizer::process(const gsl::span<o2::ctp::CTPInputDigit>
 void Digitizer::calculateClassMask(const std::bitset<CTP_NINPUTS> ctpinpmask, std::bitset<CTP_NCLASSES>& classmask)
 {
   classmask = 0;
-  for (auto const& tcl : mCTPConfiguration->getCTPClasses()) {
-    if (tcl.descriptor->getInputsMask() & ctpinpmask.to_ullong()) {
-      classmask |= (1 << tcl.classMask);
+  if (mEMCsim != 0) {
+    for (auto const& tcl : mCTPConfiguration->getCTPClasses()) {
+      if (tcl.descriptor->getInputsMask() & ctpinpmask.to_ullong()) {
+        classmask |= (1 << tcl.classMask);
+      }
+    }
+  } else {
+    for (auto const& tcl : mCTPConfiguration->getCTPClasses()) {
+      if (tcl.name.find("EMC") == std::string::npos) {
+        if (tcl.descriptor->getInputsMask() & ctpinpmask.to_ullong()) {
+          classmask |= (1 << tcl.classMask);
+          LOG(info) << "adding NOT MBA:" << tcl.name;
+        }
+      } else {
+        if ((tcl.name.find("CTVXEMC-B-NOPF-EMC") != std::string::npos) && ctpinpmask[CTP_NINPUTS - 1]) {
+          classmask |= (1 << tcl.classMask);
+          LOG(info) << "adding MBA:" << tcl.name;
+        }
+      }
```

Comment:
In this case the TVX is not checked for EMCAL (which would be OK once we use the 2-stage scenario, as the 0TVX input will be checked in the EMCAL digitization), but for the moment I would swap the order and in the outer if check for the class mask and in the inner if for the EMCAL trigger class - like this is would be always guaranteed that the trigger input TVX is applied.